flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/topic/TopicProducerRegister.java

```diff
@@ -152,7 +156,7 @@ public void close() throws IOException {
             builder.topic(topic);
             Producer<T> producer = sneakyClient(builder::create);
             producers.put(schemaInfo, producer);
-
+            pulsarSinkWriterMetrics.registerSingleProducerGauges(producer);
```

Comment:
We can add producer metrics periodically in the timeService or we can add the metrics register logic here. Not sure which one is better